pkg/controller/controller_utils.go

```diff
@@ -420,3 +421,21 @@ func FilterActivePods(pods []api.Pod) []*api.Pod {
 	}
 	return result
 }
+
+func SyncAllPodsWithStore(kubeClient client.Interface, store cache.Store) {
+	var allPods *api.PodList
+	var err error
+	listOptions := api.ListOptions{LabelSelector: labels.Everything(), FieldSelector: fields.Everything()}
+	for {
+		if allPods, err = kubeClient.Pods(api.NamespaceAll).List(listOptions); err != nil {
```

Comment:
That was the problem actually (https://github.com/kubernetes/kubernetes/issues/17829#issuecomment-161411965). That function (https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/framework/controller.go#L100) tells you if the reflector has relisted or not, but the informer still needs to take from reflector store and ADD/UPDATE/DELETE to this store via its process loop.

I want the client state to be in sync. Or is there another HasSynced function I've missed?